pkg/events/events.go

```diff
@@ -15,8 +15,9 @@ type dependencies struct {
 }
 
 type probeDependency struct {
-	Handle   probes.Handle
-	Required bool // should tracee fail if probe fails to attach
+	Handle     probes.Handle
+	Required   bool // should tracee fail if probe fails to attach
+	RootCgroup bool // cgroup probe to be attached to root cgroup hierarchy
```

Comment:
I think this should be an internal property of the `cgroupProbe` in `probes.go` not a probeDependency concern which can have many types of probes and mostly non cgroup related.
Then in the `cgroupProbe.attach` logic you can reference this field and automatically attach to the root instead of querying a root cgroup dir externally.

Reply:
Yep, I thought about it (and even coded like that in the 1st place). TBH I thought that the "cgroupProbe.attach" could be "generic" enough so user could configure where (which group directory) to attach the probe to. Lets, see..